src/context.cc

```diff
@@ -5760,16 +5795,40 @@ error Context::signupGoogle(
     const std::string &access_token,
     std::string *user_data_json,
     const uint64_t country_id) {
+    return signUpWithProvider(toggl_client, access_token, user_data_json, country_id, "", "google");
+}
+
+error Context::signupApple(
+    TogglClient *toggl_client,
+    const std::string &access_token,
+    std::string *user_data_json,
+    const std::string &full_name,
+    const uint64_t country_id) {
+    return signUpWithProvider(toggl_client, access_token, user_data_json, country_id, full_name, "apple");
+}
+
+error Context::signUpWithProvider(
+    TogglClient *toggl_client,
+    const std::string &access_token,
+    std::string *user_data_json,
+    const uint64_t country_id,
+    const std::string &full_name,
+    const std::string provider) {
     try {
         poco_check_ptr(user_data_json);
         poco_check_ptr(toggl_client);
 
         Json::Value user;
         user["google_access_token"] = access_token;
```

Comment:
In fact, we don't need to send `google_access_token` anymore since it's going to be deprecated soon. Token is required for both Apple and Google Provider

Ref: https://www.notion.so/toggl/Endpoints-9e0745dc12ca47799a209c281704fa39#6da69f9fb3fe478ca63d42be63cfdf55